airflow/www/app.py

```diff
@@ -980,25 +984,27 @@ def action(self):
                     t.task_id
                     for t in task.get_flat_relatives(upstream=True)]
             TI = models.TaskInstance
+            dates = utils.date_range(start_date, end_date)
             tis = session.query(TI).filter(
                 TI.dag_id == dag_id,
-                TI.execution_date == execution_date,
+                TI.execution_date.in_(dates),
```

Comment:
We may want to add a filter here on `not success` (I think it would reduce the number of database operations as we don't need to touch task instances that have already succeeded)